tardis/montecarlo/montecarlo_numba/base.py

```diff
@@ -128,6 +132,7 @@ def montecarlo_radial1d(
             virt_packet_last_line_interaction_out_id
         ).ravel()
     update_iterations_pbar(1)
+    refresh_packet_pbar()
```

Comment:
I agree that refreshing the progress bar after every packet makes the most logical sense. 

However, the bar is indeed updated automatically after every packet is sent. The issue arises at the end of any given iteration when `packet_pbar.update()` is called to update the packet but the change is not reflected. This can be solved by refreshing the bar at the end of every iteration (not just the final one).